janitor/functions.py

```diff
@@ -4104,3 +4104,210 @@ def process_text(
+    :param limit: number of consecutive null values to forward/backward fill.
+        Value must be greater than 0.
+    :returns: A pandas dataframe with modified column(s).
+    :raises: ValueError if the number of directions is greater than 1 and
+        the number of columns do not match the number of directions.
+    :raises: TypeError if column or direction is not of type list or tuple or
+        str.
+    """
+
+    if not columns:
+        raise ValueError("columns argument cannot be empty")
+
+    if not isinstance(columns, (list, tuple, str)):
+        raise TypeError(
+            """
+            The columns argument should be a list/tuple of column names,
+            a single column, or a string of columns separated by ','
+            """
+        )
+
+    if not isinstance(directions, (list, tuple, str)):
+        raise TypeError(
+            """
+            The directions argument should be a list/tuple of directions,
+            a single direction, or a string of directions separated by
+            ','
+            """
+        )
+
+    # if columns/directions is a string, convert to list
+    if isinstance(columns, str):
+        columns = columns.split(",")
+    if isinstance(directions, str):
+        directions = directions.split(",")
+
+    # check that all columns supplied exist in the dataframe
+    non_existent_columns = set(columns).difference(df.columns)
+    if any(non_existent_columns):
+        # handles multiple columns
+        if len(non_existent_columns) > 1:
+            outcome = [f"'{word}'" for word in non_existent_columns]
+            outcome = ", ".join(outcome)
+            raise ValueError(
+                f"Columns {outcome} do not exist in the dataframe"
+            )
+        outcome = "".join(non_existent_columns)
+        raise ValueError(f"Column '{outcome}' does not exist in the dataframe")
+
+    # check supplied directions exist in directions_options
+    direction_options = ("up", "down", "updown", "downup")
+    directions = [direction.lower() for direction in directions]
+    incorrect_direction = set(directions).difference(direction_options)
+    if any(incorrect_direction):
+        raise ValueError(
+            """
+            The direction should be either 'up', 'down', 'updown', 'downup',
+            or a combination of the aforementioned options
+            """
+        )
+
+    # check that the length of columns matches the length of directions
+    # if the length of directions is greater than 1
+    length_columns_arg = len(columns)
+    length_directions_arg = len(directions)
+
+    if length_columns_arg != length_directions_arg:
+        if length_directions_arg > 1:
+            raise ValueError(
+                """
+                The number of arguments in the columns and
+                directions do not match.
+                """
+            )
+        directions = directions * length_columns_arg
+
+    fill_pairs = dict(zip(columns, directions))
```

Comment:
With the API change, a lot of these checks can be removed. I think the only ones that would be necessary when using a dictionary are that:

- It is not an empty dictionary
- All keys exist as columns in the dataframe,
- All values are one of the four valid options.

We'd have fewer lines of code to maintain this way.